bundles/org.openhab.binding.pioneeravr/src/main/java/org/openhab/binding/pioneeravr/internal/handler/AbstractAvrHandler.java

```diff
@@ -166,16 +167,37 @@ public void handleCommand(ChannelUID channelUID, Command command) {
             boolean unknownCommand = false;
 
             if (channelUID.getId().contains(PioneerAvrBindingConstants.POWER_CHANNEL)) {
-                commandSent = connection.sendPowerCommand(command, getZoneFromChannelUID(channelUID.getId()));
+                if (command == RefreshType.REFRESH) {
+                    commandSent = connection.sendPowerQuery(getZoneFromChannelUID(channelUID.getId()));
+                } else {
+                    commandSent = connection.sendPowerCommand(command, getZoneFromChannelUID(channelUID.getId()));
+                }
             } else if (channelUID.getId().contains(PioneerAvrBindingConstants.VOLUME_DIMMER_CHANNEL)
                     || channelUID.getId().contains(PioneerAvrBindingConstants.VOLUME_DB_CHANNEL)) {
-                commandSent = connection.sendVolumeCommand(command, getZoneFromChannelUID(channelUID.getId()));
+                if (command == RefreshType.REFRESH) {
+                    commandSent = connection.sendVolumeQuery(getZoneFromChannelUID(channelUID.getId()));
+                } else {
+                    commandSent = connection.sendVolumeCommand(command, getZoneFromChannelUID(channelUID.getId()));
+                }
             } else if (channelUID.getId().contains(PioneerAvrBindingConstants.SET_INPUT_SOURCE_CHANNEL)) {
-                commandSent = connection.sendInputSourceCommand(command, getZoneFromChannelUID(channelUID.getId()));
+                if (command == RefreshType.REFRESH) {
+                    commandSent = connection.sendInputSourceQuery(getZoneFromChannelUID(channelUID.getId()));
+                } else {
+                    commandSent = connection.sendInputSourceCommand(command, getZoneFromChannelUID(channelUID.getId()));
+                }
             } else if (channelUID.getId().contains(PioneerAvrBindingConstants.LISTENING_MODE_CHANNEL)) {
-                commandSent = connection.sendListeningModeCommand(command, getZoneFromChannelUID(channelUID.getId()));
+                if (command == RefreshType.REFRESH) {
+                    commandSent = connection.sendListeningModeQuery(getZoneFromChannelUID(channelUID.getId()));
+                } else {
+                    commandSent = connection.sendListeningModeCommand(command,
+                            getZoneFromChannelUID(channelUID.getId()));
+                }
             } else if (channelUID.getId().contains(PioneerAvrBindingConstants.MUTE_CHANNEL)) {
-                commandSent = connection.sendMuteCommand(command, getZoneFromChannelUID(channelUID.getId()));
+                if (command == RefreshType.REFRESH) {
+                    commandSent = connection.sendMuteQuery(getZoneFromChannelUID(channelUID.getId()));
```

Comment:
How does the channel get refreshed though? I don't see you using the result of the query to update the state of the channel.

Reply:
The protocol allows for both device-initiated (e.g. when pressing a remote control button) as well as client-initiated (e.g. from OpenHAB) status updates.

The update message will look the same regardless of how it was triggered and in both cases the update will be handled by [StreamAvrConnection.IpControlInputStreamReader.run()](https://github.com/openhab/openhab-addons/blob/7628135323df6f07e3c2993d7b816a16e334d849/bundles/org.openhab.binding.pioneeravr/src/main/java/org/openhab/binding/pioneeravr/internal/protocol/StreamAvrConnection.java#L355) and forwared to [AbstractAvrHandler.statusUpdateReceived()](https://github.com/openhab/openhab-addons/blob/7628135323df6f07e3c2993d7b816a16e334d849/bundles/org.openhab.binding.pioneeravr/src/main/java/org/openhab/binding/pioneeravr/internal/handler/AbstractAvrHandler.java#L196) where the actual channel update will happen.